sqlmodel/_compat.py

```diff
@@ -178,6 +178,16 @@ def get_relationship_to(
         # If a list, then also get the real field
         elif origin is list:
             use_annotation = get_args(annotation)[0]
+        # If a dict or Mapping, then use the value (second) type argument
+        elif origin is dict or origin is Mapping:
+            args = get_args(annotation)
+            if len(args) >= 2:
```

Comment:
```suggestion
        elif origin is dict:
            args = get_args(annotation)
            if len(args) == 2:
```

Do we really need to support `Mapping`? If we do, we should add test for this case.
And, I think we should require exactly 2 arguments - is it valid to add more than 2 arguments to `dict` \ `Dict`?

Reply:
`typing.Dict` validates its argument count but `dict` does not.  I'll update this to check for exactly two and to remove `Mapping`